course.py

```diff
@@ -125,10 +125,8 @@ def search_teachers():
     if not validate_string_pattern(search_term):
         return make_response("Invalid search term", 400)
 
-    teachers = db.session.query(User).join(Teacher).filter(
-        User.active == True,
-        User.name.ilike(f'%{search_term}%')
-    ).all()
+    teachers = db.session.query(User).filter(User.active == True, User.is_teacher == True,
+                                             User.name.ilike(f'%{search_term}%')).all()
```

Comment:
I think it could be useful, either to : 
- specify the search is on the last name only.
- search in both name and firstname, as the complete name is chown.
- apply the `LIKE` operator on the concatenated full name
  `SELECT * FROM user WHERE CONCAT(name, ' ', first_name) LIKE '%name firs%';`